pkg/kube/client.go

```diff
@@ -232,16 +233,11 @@ func (c *Client) Update(original, target ResourceList, force bool) (*Result, err
 			c.Log("Skipping delete of %q due to annotation [%s=%s]", info.Name, ResourcePolicyAnno, KeepPolicy)
 			continue
 		}
-
-		res.Deleted = append(res.Deleted, info)
 		if err := deleteResource(info); err != nil {
-			if apierrors.IsNotFound(err) {
-				c.Log("Attempted to delete %q, but the resource was missing", info.Name)
-			} else {
-				c.Log("Failed to delete %q, err: %s", info.Name, err)
-				return res, errors.Wrapf(err, "Failed to delete %q", info.Name)
-			}
+			c.Log("Failed to delete %q, err: %s", info.ObjectName(), err)
+			continue
 		}
+		res.Deleted = append(res.Deleted, info)
```

Comment:
Nice cleanup!

For context, I think we added the `.Deleted` object beforehand because we wanted to make sure the sum of `Added + Updated + Deleted` equaled the sum of objects, but I can't be totally sure. @thomastaylor312 do you remember if there was any particular reason we pre-emptively added the objects to the `Deleted` list? If so does this look good to you?